lib/charms/prometheus_pushgateway_k8s/v0/pushgateway.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details
+
+r"""Interface library for the Prometheus Pushgateway.
+
+This library wraps the relation endpoint using the `pushgwateway` interface
+and provides a Python API for sending metrics.
+
+
+## Getting Started
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.prometheus_pushgateway_k8s/v0/pushgateway
+```
+
+In the `metadata.yaml` of the charm, add the following:
+
+```yaml
+requires:
+  pushgateway:
+    interface: pushgateway
+```
+
+In the source of your charm, first import the interface:
+
+```
+from charms.prometheus_pushgateway_k8s.v0.pushgateway import PrometheusPushgatewayInterface
+````
+
+Just instantiate the object in `__init__` and then use it when desired to send a metric, passing
+its name and value:
+
+```
+self.ppi = PrometheusPushgatewayInterface(self)
+...
+self.ppi.send_metric("test_metric", 3.141592)
+```
+
+The `send_metric` call will just end quietly if the metric was sent succesfully, or will raise
+an exception if something is wrong (that error should be logged or informed to the operator).
+
+
+## Waiting for Pushgateway
+
+When your charm is deployed but the relation is still not added to the Prometheus Pushgateway,
+metrics cannot not be sent.
+
+For robustness you should only send metrics when the interface is ready:
+
+```
+self.ppi = PrometheusPushgatewayInterface(self)
+...
+if self.ppi.is_ready:
```

Comment:
if `is_ready` is DOING stuff, such as getting state from juju (relation-get etc...) then I think it'd be more explicit to make this a classic method instead of a property. In my experience properties typically make you think that you're fetching an object's internal state.

Besides, in traefik's ingress libraries, `is_ready` is not a property, so it'd feel more consistent to have the same approach everywhere.

Reply:
Currently `is_ready` is super cheap, that's why I made it a property. But taking in consideration consistency with other libraries, and that it will start getting the relation data from juju (no cheap anymore), I made it a function.